fe/fe-core/src/main/java/org/apache/doris/catalog/Table.java

```diff
@@ -334,24 +334,48 @@ public Partition getPartition(String partitionName) {
     }
 
     public String getEngine() {
-        if (this instanceof OlapTable) {
-            return "Doris";
-        } else if (this instanceof OdbcTable) {
-            return "Odbc";
-        } else if (this instanceof MysqlTable) {
-            return "MySQL";
-        } else if (this instanceof SchemaTable) {
-            return "MEMORY";
-        } else {
-            return null;
+        switch (type) {
+            case MYSQL:
+                return "MySQL";
+            case ODBC:
+                return "Odbc";
+            case OLAP:
+                return "Doris";
+            case SCHEMA:
+                return "MEMORY";
+            case INLINE_VIEW:
+                return "InlineView";
+            case VIEW:
+                return "View";
+            case BROKER:
+                return "Broker";
+            case ELASTICSEARCH:
+                return "ElasticSearch";
+            case HIVE:
+                return "Hive";
+            default:
+                return null;
         }
     }
 
     public String getMysqlType() {
-        if (this instanceof View) {
-            return "VIEW";
+        switch (type) {
+            case OLAP:
+                return "BASE TABLE";
```

Comment:
Is it compatible with MySQL?

Reply:
MySQL has rule of "BASE TABLE for a table, VIEW for a view, or SYSTEM VIEW for an INFORMATION_SCHEMA table". And  MySQL seems not support to create external table, so add an extral "EXTERNAL TABLE".
https://dev.mysql.com/doc/refman/8.0/en/show-tables.html